pkg/collector/corechecks/snmp/internal/devicecheck/devicecheck.go

```diff
@@ -114,6 +116,13 @@ func (d *DeviceCheck) Run(collectionTime time.Time) error {
 		tags = append(tags, dynamicTags...)
 		d.sender.ServiceCheck(serviceCheckName, metrics.ServiceCheckOK, tags, "")
 	}
+	deviceReachableFloat := 0.
+	if deviceReachable {
+		deviceReachableFloat = 1
+	}
+	d.sender.Gauge(deviceReachableMetric, deviceReachableFloat, tags)
+	d.sender.Gauge(deviceUnreachableMetric, 1-deviceReachableFloat, tags)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏`1-deviceReachableFloat` feels not very readable.

```suggestion
	deviceReachableFloat := 0.
	deviceUnreachableFloat := 1.
	if deviceReachable {
		deviceReachableFloat = 1
		deviceUnreachableFloat = 0
	}
	d.sender.Gauge(deviceReachableMetric, deviceReachableFloat, tags)
	d.sender.Gauge(deviceUnreachableMetric, deviceUnreachableFloat, tags)
```

or 

```
	d.sender.Gauge(deviceReachableMetric, boolToFloat(deviceReachable), tags)
	d.sender.Gauge(deviceUnreachableMetric, boolToFloat(!deviceReachable), tags)
```
+ implement `boolToFloat()`